connectors/sources/sharepoint_online.py

```diff
@@ -673,6 +770,139 @@ async def validate_config(self):
                 f"The specified SharePoint sites [{', '.join(missing)}] could not be retrieved during sync. Examples of sites available on the tenant:[{', '.join(remote_sites[:5])}]."
             )
 
+    async def _site_access_control(self, site):
```

Comment:
I wonder why all other methods are `_with_{item}_access_control`, which delegate to `_decorate_with_access_control` in the end, and why this method and `_list_item_access_control` are special?

All the `_with_{item}_access_control` methods are used together with  `if self._dls_enabled()`, do you think we move this check inside the `_with_{item}_access_control` method? E.g.:

```python
async def _with_drive_item_access_control(self, site_drive, drive_item):
    if not self._dls_enabled():
        return drive_item

    ...
```

Reply:
Noted. Because the `site_access_control` and `list_item_access_control` are reused.